tcl/org.testeditor.tcl.dsl.tests/src/org/testeditor/tcl/dsl/jvmmodel/SimpleTypeComputerTest.xtend

```diff
@@ -0,0 +1,224 @@
+
+		// then
+		variablesWithTypes.entrySet.assertEmpty
+	}
+
+	@Test
+	def void interactionWithPrimitiveTypeParameter() {
+		// given
+		val interaction = aml.interactionTypes.findFirst[name == "wait"].assertNotNull
+
+		// when
+		val variablesWithTypes = typeComputer.getVariablesWithTypes(interaction)
+
+		// then
+		variablesWithTypes.entrySet.assertSingleElement => [
+			key.name.assertEquals("seconds")
+			value.get.qualifiedName.assertEquals('long')
+		]
+	}
+
+	@Test
+	def void macroWithNoParameter() {
+		// given
+		val macro = parseMacro('''
+			template = "Do nothing"
+		''')
+
+		// when
+		val variablesWithTypes = typeComputer.getVariablesWithTypes(macro)
+
+		// then
+		variablesWithTypes.entrySet.assertEmpty
+	}
+
+	@Test
+	def void macroWithNoParameterButInternalVariableUsage() {
+		// given
+		val macro = parseMacro('''
+			template = "Read and write"
+			Component: GreetingApplication
+			- value = Read value from <bar>
+			- Set value of <Input> to @value
+		''')
+
+		// when
+		val variablesWithTypes = typeComputer.getVariablesWithTypes(macro)
+
+		// then
+		variablesWithTypes.entrySet.assertEmpty
+	}
+
+	@Test
+	def void macroWithUnusedParameter() {
+		// given
+		val macro = parseMacro('''
+			template = "Do nothing with" ${unused}
+		''')
+
+		// when
+		val variablesWithTypes = typeComputer.getVariablesWithTypes(macro)
+
+		// then
+		variablesWithTypes.entrySet.assertSingleElement => [
+			key.name.assertEquals("unused")
+			value.assertEquals(empty)
+		]
+	}
+
+	@Test
+	def void macroWithLongParameter() {
+		// given
+		val macro = parseMacro('''
+			template = "Wait for" ${x} "seconds"
+			Component: GreetingApplication
+			- Wait for @x seconds
+		''')
+
+		// when
+		val variablesWithTypes = typeComputer.getVariablesWithTypes(macro)
+
+		// then
+		variablesWithTypes.entrySet.assertSingleElement => [
+			key.name.assertEquals("x")
+			value.get.qualifiedName.assertEquals('long')
+		]
+	}
+
+	@Test
+	def void macroWithUnusedStringAndLongParameter() {
+		// given
+		val macro = parseMacro('''
+			template = "Do nothing with" ${unused} "write" ${value} "and wait for" ${seconds} "seconds"
+			Component: GreetingApplication
+			- Set value of <Input> to @value
+			- Wait for @seconds seconds
+		''')
+
+		// when
+		val variablesWithTypes = typeComputer.getVariablesWithTypes(macro)
+
+		// then
+		variablesWithTypes.entrySet.assertSize(3) => [
+			get(0) => [
+				key.name.assertEquals('unused')
+				value.assertEquals(empty)
+			]
+			get(1) => [
+				key.name.assertEquals('value')
+				value.get.qualifiedName.assertEquals(String.name)
+			]
+			get(2) => [
+				key.name.assertEquals('seconds')
+				value.get.qualifiedName.assertEquals('long')
+			]
+		]
+	}
+
+	@Test
+	def void macroWithNestedMacroParameter() {
+		// given
+		val macro = parseMacro('''
+			template = "Wait nested for" ${x} "seconds"
+			Macro: MyMacroCollection
+			- Wait for @x seconds
```

Comment:
I'd prefer `Wait macro for @seconds` (and the template accordingly).
Then macro template and the fixture template are identical, which is a bit confusing...